### PR TITLE
[ZEE-6082] Add specification to DataProduct

### PR DESCRIPTION
**Pull request by arthurtemple**
JIRA issue: https://zeenea.atlassian.net/browse/ZEE-6082

##### PR summary:

- Ajoute `specification` à `DataProduct`

##### Checklist for Developer:

- [x] Summary part has been documented
- [x] Reviewers have been requested
- [ ] Reviews & comments have been taken into consideration
- [ ] Commits have been reworded and history cleaned

##### Checklist for Reviewer

- [ ] Code has been reviewed, commented and changes have been requested
- [ ] Reviews & comments have been taken into consideration

### CODE REVIEW

File: src/main/java/zeenea/connector/dataproduct/DataProduct.java [hunk 1]
```diff
@@ -74,7 +108,9 @@ public boolean equals(Object o) {
     DataProduct that = (DataProduct) o;
     return Objects.equals(inputPorts, that.inputPorts)
         && Objects.equals(outputPorts, that.outputPorts)
-        && Objects.equals(internalComponents, that.internalComponents);
+        && Objects.equals(internalComponents, that.internalComponents)
```
Comment (arthurtemple) [inline src/main/java/zeenea/connector/dataproduct/DataProduct.java:111]: Je ne suis pas extrêmement fan de ces méthodes `equals` car elle ne comparent pas les champs du type parent.

Comment (martin-guerre) [reply src/main/java/zeenea/connector/dataproduct/DataProduct.java:110]: J'avais corrigé les classes Dataset, Vizualisation et DataProcess sur les fonctions equals, hashcode et toString mais j'ai oublié de passer sur les classes de DataProduct :/

**Review by martin-guerre — APPROVED** [review]
Approuvé mais il faudrait corriger les fonctions `equals`, `hashCode` et `toString` des classes DataProduct

**Review by seblm — APPROVED** [review]

File: src/main/java/zeenea/connector/dataproduct/DataProduct.java (hunk 1 above)
Comment (arthurtemple) [reply src/main/java/zeenea/connector/dataproduct/DataProduct.java:96]: Je suis passé sur DataProduct.

**Review by martin-guerre — APPROVED** [review]

**Review by leborgneromain — APPROVED** [review]

**Pull request merged**